csharp/src/Apache.Arrow/Schema.cs

```diff
@@ -66,6 +61,10 @@ internal Schema(List<Field> fieldsList, IReadOnlyDictionary<string, string> meta
             _fieldsDictionary = FieldsLookup.ToDictionary(g => g.Key, g => g.First());
 
             Metadata = metadata;
+
+            _fieldsIndexLookup = _fieldsList
+                .Select((x, idx) => (Name: x.Name, Index: idx))
+                .ToLookup(x => x.Name, x => x.Index, StringComparer.CurrentCulture);
```

Comment:
I think for new code, we should use `Ordinal`. I don't think `CurrentCulture` really makes sense here.

Reply:
If I change to `Ordinal` then this would be a breaking change as this is used in `GetFieldIndex(string)`. Are you okay with chaining the behaviour?

Reply:
Of course, I can then only use the lookup if the user has provided `Ordinal` as the comparator but this would defeat the purpose of this PR which is to make lookup of Columns in RecordBatch O(1).

Reply:
Or I could change this in a separate PR that implements your suggestions here https://github.com/apache/arrow/issues/44650#issuecomment-2484205484 ?